### DISCUSSION TITLE
Infill Tools

### DISCUSSION THREAD

**Discussion by antwal** [sn4k3/UVtools#346] [Q&A]
Hi all,

the documentation in the wiki is very scarce and I would like to understand how the infill tool works as if I apply it, absolutely nothing changes.

Thanks.

**Comment by sn4k3** [comment]
If you used to FDM it work the same way. The model must be solid, if is hollow nothing will happen.
Then it will infill the body within a margin from walls (Configurable)
The density of infill is also configurable.
Use only on large models

**Reply by antwal** [reply]
Yes, I used a solid model; in prusa I only configured the external supports and the material; then i wanted to add the infill from uvtools but i couldn't do it

**Reply by antwal** [reply]
ok, i solved. 

if I select the range 40 to 1368 it didn't work well, but if I select all the layers it works.

I'm looking for the problem... 

you can add a preview of the patterns in the documentation

Thanks.

**Reply by sn4k3** [reply]
Stange thing, it should respect the range, i will take a look

**Reply by antwal** [reply]
ok i found it is a settings problem


only thing I noticed but it is another argument, by adding the supports these are also added on layer 1 where in my case there should be nothing; that is, the supports do not remain inside the object

I'll add the screen and if needed, I open an issue: 
![image](https://user-images.githubusercontent.com/410736/142470163-372820a1-b447-46d2-8484-ec12e4ad9f8c.png)

**Reply by sn4k3** [reply]
I didn't understand the last message, you can explain?

**Reply by antwal** [reply]
after adding the infill I also had to add 3 supports via the pixel editor to correct some islands that did not hook to the generated infill structure. However, these supports do not remain inside the object but have been added starting from layer 1

**Reply by sn4k3** [reply]
I still can't get it. I need a set of images or video if possible of the problem

**Reply by antwal** [reply]
ok i can prepare a video, no problems
